sql: generate flow id upfront

RaduBerinde opened the pull request:
The distsql physical planner has special logic to avoid generating a
flow UUID if everything happens to be local (unless we are doing
EXPLAIN ANALYZE). The latter is especially questionable: we are
modifying the flow specs after planning, from the `saveFlows`
callback.

This change simplifies this to always generate an UUID upfront when we
create the `PlanningCtx`. We also switch from the crypto-rand fed
`MakeV4` (~1us in the benchmarks) to `FastMakeV4()` (~40ns).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56595)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

We also should update this part https://github.com/cockroachdb/cockroach/blob/c93df89430adf2c58a26a7c1577e0ee1e08185ec/pkg/sql/colflow/vectorized_flow.go#L269-L279 since we now always have UUID set.

> Reviewed 11 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56595#-:-MLtvTVHApiKV27Melx_:b-t1kihf)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto and @miretskiy)



<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
Done, thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56595#-:-MLtwGC05_z_qGBbW2gu:b-dtkds3)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto and @miretskiy)



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56595)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto and @miretskiy)



<!-- Sent from Reviewable.io -->

asubiotto commented:
Generating UUIDs for flows in rowexec was a big hit on KV95, which is what prompted the change not to generate UUIDs for local flows. Is that not a concern anymore with `FastMakeV4`?

RaduBerinde commented:
In the UUID benchmarks on my machine, `MakeV4` is about 1100 nanoseconds and `FastMakeV4` is about 40 nanoseconds. I don't expect it to still be a problem.

RaduBerinde committed:
sql: generate flow id upfront

The distsql physical planner has special logic to avoid generating a
flow UUID if everything happens to be local (unless we are doing
EXPLAIN ANALYZE). The latter is especially questionable: we are
modifying the flow specs after planning, from the `saveFlows`
callback.

This change simplifies this to always generate an UUID upfront when we
create the `PlanningCtx`. We also switch from the crypto-rand fed
`MakeV4` (~1us in the benchmarks) to `FastMakeV4()` (~40ns).

Release note: None

RaduBerinde commented:
bors r+
